fast_pair: Add bond management documentation (#17944)
nrfconnect/sdk-nrf: doc/nrf/external_comp/bt_fast_pair.rst

You can also use the :c:func:`bt_fast_pair_has_account_key` function to check whether your Provider has any Account Keys.
This API is especially useful after a system reboot when some Account Keys may already be stored in non-volatile memory.

.. _ug_bt_fast_pair_gatt_service_bond_management:

Fast Pair bond management functionality
=======================================

To enable the Fast Pair bond management functionality, use the :kconfig:option:`CONFIG_BT_FAST_PAIR_BOND_MANAGER` Kconfig option.
When this functionality is enabled, the Fast Pair subsystem tracks the Bluetooth bonds created through the Fast Pair Procedure and unpairs them if the procedure is incomplete or the Account Key associated with the bonds is removed.
It also unpairs the Fast Pair Bluetooth bonds on Fast Pair factory reset, because the factory reset removes all Account Keys stored on device.

> See the :kconfig:option:`CONFIG_BT_FAST_PAIR_BOND_MANAGER` Kconfig option help for more datails about using the functionality.

Reviewer: Can we mention that enabling the functionality imposes some additional limitations (Fast Pair needs to be enabled while performing operations on Bluetooth bonds)? I would suggest to mention that:
`Enabling the feature imposes additional limitations related to enabling Fast Pair.`

Reviewer: The option is enabled by default for some of the use-cases. I am worried that users might overlook the limitation because of that.

The Fast Pair bond management functionality is disabled by default.

> Following use cases of the Google Fast Pair application require it to be enabled:

Reviewer 2: ```suggestion
You must enable it for the following use cases of the Google Fast Pair application:
```

Author: Changed to `It is highly recommended to enable it for the following use cases of the Google Fast Pair application:` as must was too strong.

> Following use cases of the Google Fast Pair application require it to be enabled:

Reviewer 3: Are you sure about this claim?

Reviewer: Consider using e.g. `strongly suggest` term

> It is highly recommended to enable it for the following use cases of the Google Fast Pair application:

Reviewer: The listed use-cases imply bond management. Can we rephrase a bit to clearly indicate that (current wording suggests that user needs to enable it manually)?

Author: There's a difference between the use case and the kconfig that enables the use case. User should see the kconfig to find the required feature set.

Author: From the use case section: `The selected Kconfig option configures the Fast Pair features and extensions to satisfy the `Fast Pair Device Feature Requirements`_ for your target use case.`

Author: I'd rather leave it as is. I don't want to introduce the use case Kconfigs at this point of documentation. They are introduced later.

Reviewer: Discussed offline. We will use `Make sure` syntax here.

You can also use the :c:func:`bt_fast_pair_has_account_key` function to check whether your Provider has any Account Keys.
This API is especially useful after a system reboot when some Account Keys may already be stored in non-volatile memory.

.. _ug_bt_fast_pair_gatt_service_bond_management:

Fast Pair bond management functionality
=======================================

To enable the Fast Pair bond management functionality, use the :kconfig:option:`CONFIG_BT_FAST_PAIR_BOND_MANAGER` Kconfig option.
When this functionality is enabled, the Fast Pair subsystem tracks the Bluetooth bonds created through the Fast Pair Procedure and unpairs them if the procedure is incomplete or the Account Key associated with the bonds is removed.
It also unpairs the Fast Pair Bluetooth bonds on Fast Pair factory reset, because the factory reset removes all Account Keys stored on device.

> Enabling the functionality imposes additional limitations related to enabling Fast Pair.

Reviewer: `... enabling Fast Pair in runtime.`

You can also use the :c:func:`bt_fast_pair_has_account_key` function to check whether your Provider has any Account Keys.
This API is especially useful after a system reboot when some Account Keys may already be stored in non-volatile memory.

.. _ug_bt_fast_pair_gatt_service_bond_management:

Fast Pair bond management functionality
=======================================

You can enable the :kconfig:option:`CONFIG_BT_FAST_PAIR_BOND_MANAGER` Kconfig option to enable the Fast Pair bond management functionality.
If this option is enabled, the Fast Pair subsystem tracks the Bluetooth bonds created through the Fast Pair Procedure and unpairs them if the Fast Pair Procedure is incomplete or the Account Key associated with the bonds is removed.
It also unpairs the Fast Pair Bluetooth bonds on Fast Pair factory reset.

Keep in mind that when the Fast Pair subsystem is disabled, the Bluetooth bonds are not tracked.

> Do not perform operations on the Fast Pair Bluetooth bonds apart from the Fast Pair factory reset when the Fast Pair subsystem is disabled.

Reviewer 2: ```suggestion
Do not perform operations on Fast Pair Bluetooth bonds, except for the Fast Pair factory reset when the Fast Pair subsystem is disabled.
```

> Also do not perform Bluetooth bonding when the Fast Pair subsystem is disabled and bond management functionality is enabled as it may lead to those new bonds being incorrectly considered as Fast Pair bonds by the Fast Pair subsystem.

Reviewer 2: ```suggestion
Additionally, do not perform Bluetooth bonding when the Fast Pair subsystem is disabled and bond management functionality is enabled, as this could cause the subsystem to incorrectly identify new bonds as Fast Pair bonds.
```

When this option is enabled, non-volatile storage is used to store the bond information.
You can enable this option during DFU, but you cannot disable it during DFU if you used it earlier unless you erase whole Fast Pair non-volatile storage.
Otherwise, the Fast Pair enable operation will fail.

By default, the Fast Pair bond management functionality is disabled.

> Some use cases of the Google Fast Pair application require the functionality to be enabled.

Reviewer: Please mention that use-cases that need the feature imply it.
